compiler/rustc_mir/src/borrow_check/diagnostics/mutability_errors.rs

```diff
@@ -463,6 +466,63 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
         err.buffer(&mut self.errors_buffer);
     }
 
+    fn suggest_similar_mut_method_for_for_loop(&self, err: &mut DiagnosticBuilder<'_>) {
+        let hir = self.infcx.tcx.hir();
+        let mir_hir_id = self.mir_hir_id();
+        let node = hir.item(mir_hir_id);
+        if let hir::ItemKind::Fn(_, _, body_id) = node.kind {
+            let body = hir.body(body_id);
+            if let hir::ExprKind::Block(block, _) = body.value.kind {
+                if let Some(expr) = block.expr {
+                    if let hir::ExprKind::DropTemps(expr) = expr.kind {
+                        if let hir::ExprKind::Match(expr, ..) = expr.kind {
```

Comment:
I don't understand this part. Do we need to special case this suggestion to for loops? Or do other cases already work well?

Reply:
No, we don't need to special case this suggestion in the future. But initially, this PR focuses on to suggest mutable method only for loop rhs method call.

Reply:
ok, that makes sense. Please leave a comment explaining this in the source